### DISCUSSION TITLE
More Qlik products Support

### DISCUSSION THREAD

**Discussion by mrelusive2k4** [ptarmiganlabs/butler#354] [Ideas]
Hi, Have you thought about expanding the capabilities to include triggering/chaining events from other Qlik products like NPrinting and Qlik Compose?
E.g. 
- Triggering a Qlik Nprinting task when a Qlik Sense App finishes reloading
- Triggering a Qlik Sense App reload after a Qlik Compose Datawarehouse task is complete

**Comment by mountaindude** [comment]
Indeed, those kind of automations are always interesting. 

The good news is that the various Qlik APIs usually make it possible to achieve almost anything, with the downside being that lots of API work is needed... 

The use cases you mentioned could certainly be solved using concepts already existing in Butler. 

- Triggering a Qlik Nprinting task when a Qlik Sense App finishes reloading.
   Could be done several ways
   - Forward log events from Sense Butler when reloads finish, then have a lookup table in Butler that controls which NP jobs to start. Or set that using custom properties in the QMC, for the reload task (or app) in question.
   - Subscribe to notifications from the engine. This is in my experience not as robust (for various rather technical reasons) as the previous option.
- Triggering a Qlik Sense App reload after a Qlik Compose Datawarehouse task is complete
   - I haven't worked much with Compose so I'm not sure what options there are to detect that a Compose task has completed. I'd be surprised if it wasn't possible though...


Finally, as a generic solution it's also possible to use a messaging solution like MQTT (light weight and open source, but the messaging solutions from the big clouds also work for this, of course) as middleware. 
The source system would then publish a message "I am done", Butler would read that and know what to do with it (i.e. trigger some action in QS, NP etc). Classic messaging concept.

**Comment by mountaindude** [comment]
While very good ideas, this kind of scheduling is not currently in scope of the Butler project.
